lib/ansible/module_utils/vmware.py

```diff
@@ -648,3 +649,109 @@ def find_host_by_cluster_datacenter(module, content, datacenter_name, cluster_na
             return host, cluster
 
     return None, cluster
+
+
+def set_vm_power_state(content, vm, state, force):
+    """
+    Set the power status for a VM determined by the current and
+    requested states. force is forceful
+    """
+    facts = gather_vm_facts(content, vm)
+    expected_state = state.replace('_', '').lower()
```

Comment:
The expected_state variable only strips the underscore (_) character from the state variable, but the providable power states for vmware_guest_powerstate use hyphens (-), so the expected state will never match the current_state or any of the other conditionals.  

Using underscores for the state names (ex. powered_off) strikes me as being more consistent with the existing conventions than adding hyphens.